lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc

```diff
@@ -283,18 +286,33 @@ function dosomething_reportback_form_save_file(&$form_state) {
     $y = $form_state['values']['crop_y'];
     $width = $form_state['values']['crop_width'];
     $height = $form_state['values']['crop_height'];
+    $rotate = $form_state['values']['crop_rotate'];
     // Load the image.
     $image = image_load($file->uri);
-    // Crop the image.
-    dosomething_reportback_image_crop($image, $x, $y, $width, $height);
+    // Apply user crop and rotation to the image.
+    dosomething_reportback_image_edit($image, $x, $y, $width, $height, $rotate);
   }
   return $file;
 }
 
-// Function to crop image.
-function dosomething_reportback_image_crop($image, $x, $y, $width, $height) {
-  // Crop the image.
-  $result = image_crop($image, $x, $y, $width, $height);
+/**
+ * Rotate and crops an image based on user input.
+ */
+function dosomething_reportback_image_edit($image, $x, $y, $width, $height, $rotate) {
+  // Rotate the image and then crop it if rotation was successful.
+  if ($rotate) {
+    $result = image_rotate($image, $rotate);
+
+    // Crop the image.
+    if ($result) {
+      $result = image_crop($image, $x, $y, $width, $height);
+    }
+  }
+  // If rotation not defined, just crop the image.
+  else {
+    $result = image_crop($image, $x, $y, $width, $height);
+  }
+
```

Comment:
@weerd @DFurnes Hmm yea I had it like that before but then I was thinking. What if rotation failed for some reason? If we try to rotate and it fails, `$result` would be false but we would still try to crop it which would then either also fail (worst case) or it would crop the image without the rotation. It's not the worst out come but what do you think?